spec/unit/ssl/certificate_request_spec.rb

```diff
@@ -177,6 +177,17 @@
       end
     end
 
+    context "with DNS and IP SAN specified" do
+      before :each do
+        Puppet[:dns_alt_names] = ""
```

Comment:
Do we want to take this opportunity to rename/alias the setting as well?

Reply:
Given the use case described in the docs here https://puppet.com/docs/puppetserver/5.3/scaling_puppet_server.html#creating-and-configuring-compile-masters I'm inclined to say yes.

Reply:
The definition of the `dns_alt_names` setting can be found here https://github.com/puppetlabs/puppet/blob/master/lib/puppet/defaults.rb#L745. There's a big doc there that at minimum should be updated to describe the new tagged syntax. I think it would be good to also deprecate that setting in favor of one called `subject_alt_names`. For an example of a deprecated setting, see https://github.com/puppetlabs/puppet/blob/master/lib/puppet/defaults.rb#L1814-L1823.